File: FFTTextField/ViewController.m
//
//  ViewController.m
//  FFTTextField
//
//  Created by 四五20 on 2019/7/18.
//  Copyright © 2019 四五20. All rights reserved.
//

#import "ViewController.h"
#import "UITextField+FFT_Category.h"
#import <objc/runtime.h>

@interface ViewController ()
<
    UITextFieldDelegate
>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    UITextField *fd = [UITextField new];
    
    fd.placeholder = @"please";
    
    fd.delegate = self;
    
    fd.fft_max = 4;
    
    [self.view addSubview:fd];
    
    fd.frame = CGRectMake(100, 100, 200, 30);
    
    fd.borderStyle = UITextBorderStyleRoundedRect;
}

- (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string {
    return YES;
}

@end

File: FFTTextField-Swift/FFTTextField-Swift/ViewController.swift
//
//  ViewController.swift
//  FFTTextField-Swift
//
//  Created by 四五20 on 2019/7/23.
//  Copyright © 2019 四五20. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITextFieldDelegate {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.loadUI()
    }

    func loadUI() {
        self.view.backgroundColor = UIColor.white
        
        let textField = UITextField.init(frame: CGRect(x:100, y:100, width:100, height:30))
        
        textField.borderStyle = UITextField.BorderStyle.roundedRect
        
        textField.delegate = self
        
        textField.fft_max = 5
        
        textField.fft_inputType = .number
        
        view.addSubview(textField)
    }
    
    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
        return true
    }
    
}

